_posts/2020-10-05-torch-network-with-autograd/torch-network-with-autograd.Rmd

```diff
@@ -0,0 +1,347 @@
+---
+title: "Introducing torch autograd"
+description: > 
+ Instructive though it may be to code backpropagation from scratch, with torch, there is hardly ever a reason to. Its automatic differentiation feature, called autograd, keeps track of operations that need their gradients computed, as well as how to compute them.
+ In this second post of a four-part series, we update our simple, hand-coded network to make use of autograd.
+author:
+  - name: Sigrid Keydana
+    affiliation: RStudio
+    affiliation_url: https://www.rstudio.com/
+slug: keydanatorchnetworkwithautograd
+date: 10-05-2020
+categories:
+  - Torch
+  - R
+output:
+  distill::distill_article:
+    self_contained: false
+    toc: true
+preview: images/preview.jpg
+editor_options: 
+  markdown: 
+    wrap: 72
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, eval = FALSE)
+```
+
+Last week, we saw how to code [a simple network from
+scratch](https://blogs.rstudio.com/ai/posts/2020-10-01-torch-network-from-scratch),
+using nothing but `torch` *tensors*. Predictions, loss, gradients,
+weight updates -- all these things we've been computing ourselves.
+Today, we make a significant change: Namely, we spare ourselves the
+cumbersome calculation of gradients, and have `torch` do it for us.
+
+Prior to that though, let's get some background.
+
+## Automatic differentiation with *autograd*
+
+`Torch` uses a module called *autograd* to
```

Comment:
Is this torch the package? If so, it should remain lowercase, since R is case-sensitive.

Reply:
Well, I capitalized it because beginning-of-sentence ... but will change if you say so :-)